[BUG] Main build broken — missing ensureVoiceTurn/clearPendingDone in GigiSmartOrchestrator after df5a645

## 🎯 Cosa sta succedendo

Main non compila più (`xcodebuild -configuration Debug` su `generic/platform=iOS`). Bloccante: **nessuno può buildare nessuna feature branch** finché non torna verde. Scoperto durante build verify di sub-issue #60 (empty speech guard).

## 🔧 Root cause

Commit [`df5a645`](https://github.com/Building-addicts/GIGI/commit/df5a645) ("feat(ios): saga del wake-word, della Live Activity e dell'orchestrazione audio") di @Leonardo-Corte ha introdotto in `GigiSmartOrchestrator.swift` chiamate a metodi che **non esistono nel repo**:

- `ensureVoiceTurn` — chiamato a righe 163, 352, 373, 384
- `clearPendingDone` — chiamato a riga 385

`grep -r "func ensureVoiceTurn"` su tutto `02_GIGI_APP/` → **zero match**. Sospetto: il file che definisce questi helper non è stato committato (resto sul Mac di Leo) oppure rimosso per sbaglio.

Inoltre — meno grave, ma collegato — 11 call a `GigiDebugLogger.voiceEvent(name, nil, dict)` passano `nil` posizionale dove la signature richiede label `turnId:`. Distribuiti in:
- `GigiAudioManager.swift` — 5 call (righe 158, 190, 197, 214, 244)
- `GigiSmartOrchestrator.swift` — 6 call (righe 157, 164, 251, 265, 274, 386)

## ✨ Risultato atteso

`xcodebuild ... build` torna `BUILD SUCCEEDED` su main. Sblocca tutte le feature branch (#60, #50/51/52, #61-#63, ecc).

## File toccati attesi

- `02_GIGI_APP/GIGI/GigiSmartOrchestrator.swift` — aggiungere definizioni `ensureVoiceTurn` e `clearPendingDone` + label `turnId:` a voiceEvent
- `02_GIGI_APP/GIGI/GigiAudioManager.swift` — label `turnId:` a 5 voiceEvent
- (probabilmente) un file mancante con extension `GigiSmartOrchestrator+VoiceTurn.swift` o simile, da recuperare dal Mac di Leo

## Errori completi

```
GigiSmartOrchestrator.swift:163:22: error: cannot find 'ensureVoiceTurn' in scope
GigiSmartOrchestrator.swift:352:13: error: cannot find 'ensureVoiceTurn' in scope
GigiSmartOrchestrator.swift:373:13: error: cannot find 'ensureVoiceTurn' in scope
GigiSmartOrchestrator.swift:384:22: error: cannot find 'ensureVoiceTurn' in scope
GigiSmartOrchestrator.swift:385:9:  error: cannot find 'clearPendingDone' in scope
GigiSmartOrchestrator.swift:157:72, 251:73, 265:61, 274:65: cannot convert 'String?' to '[String:String]'
GigiSmartOrchestrator.swift:164:35, 274:35, 386:35: missing argument labels 'turnId:location:function:line:' in call
GigiAudioManager.swift:158, 190, 197, 214, 244: missing argument labels 'turnId:'
** BUILD FAILED **
```

## Fix parziale già pushato

Branch [`fix/audiomgr-voiceevent-labels`](https://github.com/Building-addicts/GIGI/tree/fix/audiomgr-voiceevent-labels) commit `00eb654` — fixa solo le 5 label di `GigiAudioManager.swift`. Riusabile come componente del fix completo. **Da NON mergiare da solo** (insufficiente).

## Acceptance criteria

- [ ] `xcodebuild -project GIGI.xcodeproj -scheme GIGI -configuration Debug -destination 'generic/platform=iOS' CODE_SIGN_IDENTITY='' CODE_SIGNING_REQUIRED=NO CODE_SIGNING_ALLOWED=NO build` ritorna `BUILD SUCCEEDED`
- [ ] Nessun `error:` nei log
- [ ] Comportamento runtime invariato per voice turn lifecycle (verifica test E2E base: turno wake → listening → thinking → speaking → done)

## Impatto / urgenza

🚨 **Release blocker.** Code freeze mercoledì 30 (oggi è 27 aprile). Demo venerdì 1 maggio. Ogni dev su feature branch è bloccato dal poter testare. La sub-issue #60 (empty speech guard, parte del parent #16 Resilience) è in stand-by fino a fix.

cc @ArmandoBattaglino @Leonardo-Corte

Riferimenti:
- Commit colpevole: df5a645
- Branch fix parziale: `fix/audiomgr-voiceevent-labels`
- Sub-issue bloccata: #60 (parent #16)

--- a/02_GIGI_APP/GIGI/GigiAudioManager.swift
+++ b/02_GIGI_APP/GIGI/GigiAudioManager.swift
@@ -155,7 +155,7 @@ final class GigiAudioManager: ObservableObject {
         if suppressNextSpeakingFinished {
             suppressNextSpeakingFinished = false
             if state == .speaking { transition(to: .idle) }
-            GigiDebugLogger.voiceEvent("audio.speakingFinishSuppressed", nil, dataForTrace())
+            GigiDebugLogger.voiceEvent("audio.speakingFinishSuppressed", turnId: nil, dataForTrace())
             print("GigiAudioManager: speaking finish suppressed for barge-in")
             return
         }
@@ -187,14 +187,14 @@ final class GigiAudioManager: ObservableObject {
 
     private func scheduleFollowUpTimeout() {
         followUpTimeoutTask?.cancel()
-        GigiDebugLogger.voiceEvent("audio.followUpWindowStarted", nil, dataForTrace())
+        GigiDebugLogger.voiceEvent("audio.followUpWindowStarted", turnId: nil, dataForTrace())
         followUpTimeoutTask = Task { @MainActor [weak self] in
             try? await Task.sleep(nanoseconds: UInt64(Self.presenceFollowUpWindow * 1_000_000_000))
             guard !Task.isCancelled, let self else { return }
             // User actually spoke → VAD fired → state already moved off .recording. No-op.
             guard self.state == .recording else { return }
             print("GigiAudioManager: presence follow-up window elapsed — back to wake-word standby")
-            GigiDebugLogger.voiceEvent("audio.followUpWindowElapsed", nil, self.dataForTrace())
+            GigiDebugLogger.voiceEvent("audio.followUpWindowElapsed", turnId: nil, self.dataForTrace())
             self.stopRecording()
             self.resumeWakeWordIfEnabled()
         }
@@ -211,7 +211,7 @@ final class GigiAudioManager: ObservableObject {
 
     private func transition(to newState: GigiAudioState) {
         print("GigiAudioManager: \(state) → \(newState)")
-        GigiDebugLogger.voiceEvent("audio.transition", nil, ["from": "\(state)", "to": "\(newState)", "presenceMode": "\(presenceMode)"])
+        GigiDebugLogger.voiceEvent("audio.transition", turnId: nil, ["from": "\(state)", "to": "\(newState)", "presenceMode": "\(presenceMode)"])
         // Any transition out of .recording invalidates the Presence follow-up window.
         if state == .recording, newState != .recording {
             followUpTimeoutTask?.cancel()
@@ -241,7 +241,7 @@ final class GigiAudioManager: ObservableObject {
 
     private func resumeWakeWordIfEnabled() {
         print("GigiAudioManager: resumeWakeWordIfEnabled — presence=\(presenceMode)")
-        GigiDebugLogger.voiceEvent("audio.resumeWakeWordIfEnabled", nil, dataForTrace())
+        GigiDebugLogger.voiceEvent("audio.resumeWakeWordIfEnabled", turnId: nil, dataForTrace())
         if presenceMode { startWakeWordListening() }
     }
 

--- a/02_GIGI_APP/GIGI/GigiSmartOrchestrator.swift
+++ b/02_GIGI_APP/GIGI/GigiSmartOrchestrator.swift
@@ -154,14 +154,14 @@ class GigiSmartOrchestrator: ObservableObject {
         stopMicCapture()
         let trimmed = text.trimmingCharacters(in: .whitespacesAndNewlines)
         guard !trimmed.isEmpty else {
-            GigiDebugLogger.voiceEvent("orchestrator.emptyTranscript", currentVoiceTurnId)
+            GigiDebugLogger.voiceEvent("orchestrator.emptyTranscript", turnId: currentVoiceTurnId)
             isThinking = false
             currentVoiceTurnId = nil
             return
         }
 
         let turnId = ensureVoiceTurn(reason: "transcript")
-        GigiDebugLogger.voiceEvent("orchestrator.transcript", turnId, ["length": "\(trimmed.count)"])
+        GigiDebugLogger.voiceEvent("orchestrator.transcript", turnId: turnId, ["length": "\(trimmed.count)"])
 
         await GigiLiveActivityController.shared.transitionToThinking(transcript: trimmed)
         status = "GIGI: Sto pensando..."
@@ -248,7 +248,7 @@ class GigiSmartOrchestrator: ObservableObject {
     // MARK: - Deferred turn close (T4)
 
     private func scheduleDoneAfterTTS(message: String) {
-        GigiDebugLogger.voiceEvent("orchestrator.scheduleDoneAfterTTS", currentVoiceTurnId)
+        GigiDebugLogger.voiceEvent("orchestrator.scheduleDoneAfterTTS", turnId: currentVoiceTurnId)
         pendingDoneMessage = message
         doneSafetyTask?.cancel()
         // Safety: if AVSpeechSynthesizer never reports finish (cancel storms, hardware
@@ -262,7 +262,7 @@ class GigiSmartOrchestrator: ObservableObject {
 
     private func fireDone() {
         guard let msg = pendingDoneMessage else { return }
-        GigiDebugLogger.voiceEvent("orchestrator.fireDone", currentVoiceTurnId)
+        GigiDebugLogger.voiceEvent("orchestrator.fireDone", turnId: currentVoiceTurnId)
         pendingDoneMessage = nil
         doneSafetyTask?.cancel()
         doneSafetyTask = nil
@@ -271,7 +271,7 @@ class GigiSmartOrchestrator: ObservableObject {
     }
 
     private func finalizeTurnNow(message: String) {
-        GigiDebugLogger.voiceEvent("orchestrator.finalizeTurn", currentVoiceTurnId, ["presenceMode": "\(GigiAudioManager.shared.presenceMode)", "quickTalk": "\(isQuickTalkSession)"])
+        GigiDebugLogger.voiceEvent("orchestrator.finalizeTurn", turnId: currentVoiceTurnId, ["presenceMode": "\(GigiAudioManager.shared.presenceMode)", "quickTalk": "\(isQuickTalkSession)"])
         SoundEngine.releaseSession()
 
         if isQuickTalkSession {
@@ -383,7 +383,7 @@ class GigiSmartOrchestrator: ObservableObject {
     func interruptAndListen(source: String) {
         let turnId = ensureVoiceTurn(reason: "interrupt.\(source)")
         clearPendingDone(reason: "bargeIn.\(source)")
-        GigiDebugLogger.voiceEvent("orchestrator.interruptAndListen", turnId, ["source": source, "audioState": "\(GigiAudioManager.shared.state)"])
+        GigiDebugLogger.voiceEvent("orchestrator.interruptAndListen", turnId: turnId, ["source": source, "audioState": "\(GigiAudioManager.shared.state)"])
 
         SoundEngine.play(.wakeWord)
         if isQuickTalkSession { onQuickTalkStateChange?(.listening) }
@@ -459,5 +459,35 @@ class GigiSmartOrchestrator: ObservableObject {
         }
 
         return validParts.count >= 2 ? validParts : nil
+    }
+
+    // MARK: - Voice turn lifecycle helpers
+    //
+    // Reconstructed from call-site contracts after #95 (ensureVoiceTurn /
+    // clearPendingDone were referenced by `df5a645` but their definitions
+    // were never committed). Behaviour kept conservative: log + minimal
+    // state mutation, no impact on existing turn flow.
+
+    /// Returns the current voice turn id, generating a new one if none is active.
+    /// `reason` is a short tag describing the trigger (transcript / wake / interrupt / quickTalk).
+    @discardableResult
+    fileprivate func ensureVoiceTurn(reason: String) -> String {
+        if let existing = currentVoiceTurnId {
+            GigiDebugLogger.voiceEvent("orchestrator.ensureVoiceTurn.reuse", turnId: existing, ["reason": reason])
+            return existing
+        }
+        let newId = String(UUID().uuidString.prefix(8))
+        currentVoiceTurnId = newId
+        GigiDebugLogger.voiceEvent("orchestrator.ensureVoiceTurn.new", turnId: newId, ["reason": reason])
+        return newId
+    }
+
+    /// Cancels any pending deferred-done state (used on barge-in / interrupt).
+    fileprivate func clearPendingDone(reason: String) {
+        guard pendingDoneMessage != nil || doneSafetyTask != nil else { return }
+        GigiDebugLogger.voiceEvent("orchestrator.clearPendingDone", turnId: currentVoiceTurnId, ["reason": reason])
+        pendingDoneMessage = nil
+        doneSafetyTask?.cancel()
+        doneSafetyTask = nil
     }
 }

